### PR TITLE
Update API examples

### PR DESCRIPTION
**Pull request by divyam3897**
Improve api examples.

## PR Summary
Removes the redundant keywords in some of api examples and improve them.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ X] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: examples/api/custom_projection_example.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 Custom projection
 =================
 
-This example showcases the Hammer projection by alleviating many features of
+Hammer projection by alleviating many features of
```
Comment (QuLogic) [inline examples/api/custom_projection_example.py]: This sentence fragment no longer makes sense like this. Though to be honest, I don't think "alleviating" was the right word in the first place.

File: examples/api/filled_step.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 Hatch-filled histograms
 =========================
 
-This example showcases the hatching capabilities of matplotlib by plotting
+Hatching capabilities of matplotlib by plotting
```
Comment (QuLogic) [inline examples/api/filled_step.py]: Sentence fragment doesn't work anymore.

File: examples/api/histogram_path.py [hunk 3]
```diff
@@ -3,8 +3,8 @@
 Building histograms using Rectangles and PolyCollections
 ========================================================
 
-This example shows how to use a path patch to draw a bunch of
-rectangles.  The technique of using lots of Rectangle instances, or
+A path patch to draw a bunch of rectangles.
```
Comment (QuLogic) [inline examples/api/histogram_path.py]: Sentence fragment does not make sense.

File: examples/api/colorbar_basics.py [hunk 4]
```diff
@@ -3,7 +3,7 @@
 Colorbar
 ========
 
-This example shows how to use colorbar by specifying the mappable object (here
```
Comment (dstansby) [inline examples/api/colorbar_basics.py]: 👍

File: examples/api/histogram_path.py (hunk 3 above)
Comment (dstansby) [reply examples/api/histogram_path.py]: I think it does here

File: examples/api/legend.py [hunk 5]
```diff
@@ -3,7 +3,7 @@
 Legend using pre-defined labels
 ===============================
 
-Notice how the legend labels are defined with the plots!
+Legend labels defined with the plots.
```
Comment (dstansby) [inline examples/api/legend.py]: How about "Defining legend labels with plots"?

File: examples/api/filled_step.py (hunk 2 above)
Comment (dstansby) [reply examples/api/filled_step.py]: How about "Hatching capabilities for plotting histograms"?

**Comment by tacaswell** [comment]
@divyam3897 Changed the title because the prefix `API:` typically means that the API is being changed.  This way the title is much less alarming :stuck_out_tongue_winking_eye: .

File: examples/api/histogram_path.py (hunk 3 above)
Comment (QuLogic) [reply examples/api/histogram_path.py]: It might have worked if it were the only thing, but right now, it's just a subject with no verb.

**Comment by divyam3897** [comment]
@dstansby @QuLogic does it look better now? :)

File: examples/api/histogram_path.py (hunk 3 above)
Comment (jklymak) [reply examples/api/histogram_path.py]: Suggest:   Using a path patch to draw rectangles.

**Comment by divyam3897** [comment]
@jklymak done 👍

**Comment by jklymak** [comment]
@dstansby, this now OK w/ you?

**Pull request merged**